internal/services/controllers/jobs/queue.go

```diff
@@ -356,6 +375,64 @@ func (q *queue) processStepRunUpdates(ctx context.Context, tenantId string) (boo
 	return res.Continue, nil
 }
 
+func (q *queue) runTenantUpdateStepRunsV2(ctx context.Context) func() {
+	return func() {
+		q.l.Debug().Msgf("partition: updating step run statuses (v2)")
+
+		// list all tenants
+		tenants, err := q.repo.Tenant().ListTenantsByControllerPartition(ctx, q.p.GetControllerPartitionId())
+
+		if err != nil {
+			q.l.Err(err).Msg("could not list tenants")
+			return
+		}
+
+		q.updateStepRunV2Operations.SetTenants(tenants)
+
+		for i := range tenants {
+			tenantId := sqlchelpers.UUIDToStr(tenants[i].ID)
+
+			q.updateStepRunV2Operations.RunOrContinue(tenantId)
+		}
+	}
+}
+
+func (q *queue) processStepRunUpdatesV2(ctx context.Context, tenantId string) (bool, error) {
+	ctx, span := telemetry.NewSpan(ctx, "process-step-run-updates-v2")
+	defer span.End()
+
+	dbCtx, cancel := context.WithTimeout(ctx, 300*time.Second)
```

Comment:
intentional 300 sec?

Reply:
I think I left it the same as ProcessStepRunUpdates, I think I just left this intentionally high before.